### PR TITLE
Fix generated Equals(object) parameter nullability mismatch

### LINKED ISSUE #38068: Generated Equals(object) signature does not match nullability of base method
**Version Used**: VS 16.3 Preview 2

Generate equality members for:

```cs
struct Foo
{
    public int Bar { get; }
}
```

Result:

```cs
    // CS8610 Nullability of reference types in type of parameter 'obj' doesn't match overridden
    // member.           ↓
    public override bool Equals(object obj)
    {
        return obj is Foo foo && Equals(foo);
    }
```

### PR DESCRIPTION
**Pull request by NextTurn**
Fixes #38068

This fixes the Quick Action. What about the `Equal(object)` methods for anonymous types?

### CODE REVIEW

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
                 explicitInterfaceImplementations: default,
                 name: EqualsName,
                 typeParameters: default,
-                parameters: ImmutableArray.Create(CodeGenerationSymbolFactory.CreateParameterSymbol(compilation.GetSpecialType(SpecialType.System_Object), ObjName)),
+                parameters: ImmutableArray.Create(CodeGenerationSymbolFactory.CreateParameterSymbol(compilation.GetSpecialType(SpecialType.System_Object).WithNullability(NullableAnnotation.Annotated), ObjName)),
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs]: canyou add a test validating any change in behavior?

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs [hunk 2]
```diff
@@ -22,6 +22,9 @@ public class GenerateEqualsAndGetHashCodeFromMembersTests : AbstractCSharpCodeAc
         private static readonly TestParameters CSharp6 =
             new TestParameters(parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6));
 
+        private static readonly TestParameters CSharp8 =
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs]: This should be the automatic option if nothing is specified; usually we wouldn't say it so we still have coverage once 8.1 or 9.0 or whatever is a thing.

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs [hunk 3]
```diff
@@ -398,6 +401,31 @@ public override bool Equals(object obj)
...
+@"#nullable enable
+
+class Program
+{
+    int a;
+
+    public override bool Equals(object? obj)
+    {
+        return obj is Program program &&
+               a == program.a;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs]: thanks!

**Comment by jasonmalinowski** [comment]
Actually merging this now since I'm fixing the other bug around GetHashCode and I need this in too. :-) Thanks @NextTurn!

**Pull request merged**